src/wp-admin/includes/image.php

```diff
@@ -900,7 +900,6 @@ function wp_generate_attachment_metadata( $attachment_id, $file ) {
 				$preview_file = $dirname . wp_unique_filename( $dirname, wp_basename( $file, $ext ) . '-pdf.jpg' );
```

Comment:
This block assumes the primary type is jpeg, lets stick that with a mapping in `wp_upload_image_mime_transforms` so this first full size image is the primary type from that (eg. it could be ONLY WebP)

Reply:
Added changes in latest commit.